test/crash/src/test/java/io/narayana/lra/arquillian/LRACoordinatorRecoveryIT.java

```diff
@@ -150,57 +151,24 @@ public static WebArchive deploy() {
     public void lraCoordinatorShortTimeoutLRA(@ArquillianResource @OperateOnDeployment(LRA_PARTICIPANT_DEPLOYMENT_QUALIFIER) URL baseURL)
             throws URISyntaxException {
 
-        String lraId;
-        URI lraListenerURI = UriBuilder.fromUri(baseURL.toURI()).path(LRAListener.LRA_LISTENER_PATH).build();
-
-        // Starts an LRA with a short time limit by invoking a resource annotated with @LRA. The time limit is
-        // defined as LRAListener.LRA_SHORT_TIMELIMIT
-        Response response = client
-                .target(lraListenerURI)
-                .path(LRAListener.LRA_LISTENER_ACTION)
-                .request()
-                .put(null);
-
-        Assert.assertEquals("LRA participant action", 200, response.getStatus());
+        Long SHORTER_TIMEOUT = 1000L;
+        URI shortLRA = lraClient.startLRA(null, "Short Timeout Recovery Test", SHORTER_TIMEOUT, ChronoUnit.MILLIS);
 
+        // The LRA transaction could have been started via lraClient but it is useful to
+        // test the filters as well
         // Restarts lra-coordinator to simulate a crash
         restartContainer(LRA_COORDINATOR_CONTAINER_QUALIFIER);
 
         // Waits for a period of time longer than the timeout of the LRA Transaction
-        doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
-        // The LRA transaction could have been started via lraClient but it is useful to test the filters as well
-        lraId = getFirstLRAFromFS();
-        assertNotNull("A new LRA should have been added to the object store before the JVM was halted.", lraId);
-        lraId = String.format("%s/%s", lraClient.getCoordinatorUrl(), lraId);
-
-        // Waits for a period of time longer than the timeout of the LRA Transaction
-        doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
-        // Checks recovery
-        LRAStatus status = getStatus(new URI(lraId));
-
-        LRALogger.logger.infof("%s: Status after restart is %s%n", status == null ? "GONE" : status.name());
-
-        if (status == null || status == LRAStatus.Cancelling) {
-            int sc = recover();
-
-            if (sc != 0) {
-                recover();
-            }
-        }
+        doWait(SHORTER_TIMEOUT + 1000L);
 
         // LRA with short timeout should have timed out and cancelled
-        status = getStatus(new URI(lraId));
+        LRAStatus status = getStatus(shortLRA);
 
-        Assert.assertTrue(String.format("LRA %s should have cancelled", lraId),
-                status == null || status == LRAStatus.Cancelled);
 
-        // Verifies that the resource was notified that the LRA finished
-        String listenerStatus = getStatusFromListener(lraListenerURI);
```

Comment:
So changing the test to not using the lraListenerURI has removed the ability to test this part. Do we have another other test of retrieving a status from a Listener?

Reply:
I've added a test for that. Thanks